test/integration/sequelize.test.js

```diff
@@ -728,6 +728,23 @@ describe(Support.getTestDialectTeaser('Sequelize'), () => {
       });
     });
 
+    it('connections should be released only once', function() {
+      return this.sequelize.query('THIS IS A WRONG SQL', {
+        retry: {
+          max: 2,
+          // retry for all errors
+          match: null
+        }
+      }).catch(err => {
+        // If the connection is released multiple times,
+        // than an Error(not Sequelize BaseError) will be raised by generic-pool.
+        // If the connection is released only once,
+        // then a DatabaseError which inherits Sequelize BaseError will be raised,
+        // because this is a wrong sql.
+        expect(err).to.be.an.instanceof(this.sequelize.Error);
```

Comment:
Add a spy on `this.sequelize.connectionManager.releaseConnection`, assert against number of calls